openpype/hosts/blender/api/utils.py

```diff
@@ -574,3 +574,290 @@ def get_used_datablocks(
+                == d.get("source_name")
+            ),
+            None,
+        ):
+            old_datablock.user_remap(new_datablock)
+
+            # Remove remapped datablock
+            datablocks_to_remap.remove(new_datablock)
+
+            # Skip if pure link because changes won't be saved
+            if new_datablock.library:
+                continue
+
+            # Transfer transforms
+            if isinstance(old_datablock, bpy.types.Object):
+                new_datablock.location = old_datablock.location
+                new_datablock.rotation_euler = old_datablock.rotation_euler
+                new_datablock.scale = old_datablock.scale
+
+            # Ensure action relink
+            if (  # Object
+                hasattr(old_datablock, "animation_data")
+                and old_datablock.animation_data
+            ):
+                transfer_action(old_datablock, new_datablock)
+            if (  # Object data (ie mesh, light...)
+                hasattr(old_datablock, "data")
+                and hasattr(old_datablock.data, "animation_data")
+                and old_datablock.data.animation_data
+            ):
+                transfer_action(old_datablock.data, new_datablock.data)
+
+            # Ensure modifiers reassignation
+            if hasattr(old_datablock, "modifiers"):
+                transfer_stack(old_datablock, "modifiers", new_datablock)
+
+            # Ensure constraints reassignation
+            if hasattr(old_datablock, "constraints"):
+                transfer_stack(old_datablock, "constraints", new_datablock)
+
+            # Ensure bones constraints reassignation
+            if hasattr(old_datablock, "pose") and old_datablock.pose:
+                for bone in old_datablock.pose.bones:
+                    if new_datablock.pose:
+                        if new_bone := new_datablock.pose.bones.get(
+                            bone.name
+                        ):
+                            transfer_stack(bone, "constraints", new_bone)
+
+            # Ensure drivers reassignation
+            if (
+                isinstance(old_datablock, bpy.types.Object)
+                and hasattr(new_datablock.data, "shape_keys")
+                and new_datablock.data.shape_keys
+                and new_datablock.data.shape_keys.animation_data
+                and old_datablock.data
+                and old_datablock.data.shape_keys
+                and old_datablock.data.shape_keys.animation_data
+                and old_datablock.data.shape_keys.animation_data.drivers
+            ):
+                for i, driver in enumerate(
+                    new_datablock.data.shape_keys.animation_data.drivers
+                ):
+                    for j, var in enumerate(driver.driver.variables):
+                        for k, target in enumerate(var.targets):
+                            target.id = (
+                                old_datablock.data.shape_keys.animation_data.drivers[
```

Comment:
Line too long.

Reply:
Yes but that's how black formats it

Reply:
```suggestion
                                old_datablock
                                .data
                                .shape_keys
                                .animation_data
                                .drivers[
```
If this is meant to be merged upstream, they'll check for linting and this won't pass.